core/state_transition.go

```diff
@@ -432,6 +448,39 @@ func (st *StateTransition) refundGas(apricotPhase1 bool) {
 	st.gp.AddGas(st.gas)
 }
 
+func stateConnectorContract(chainID *big.Int, blockTime *big.Int) common.Address {
+	switch {
+	case isStateConnectorActivated(chainID, blockTime) && chainID.Cmp(songbirdChainID) == 0:
+		return songbirdStateConnectorAddress
+	default:
+		return common.HexToAddress("0x1000000000000000000000000000000000000001")
+	}
+}
+
+func isStateConnectorActivated(chainID *big.Int, blockTime *big.Int) bool {
+	switch {
+	case isTestingChain(chainID):
+		return true
+	case chainID.Cmp(flareChainID) == 0:
+		return blockTime.Cmp(flareStateConnectorActivationTime) >= 0
+	case chainID.Cmp(songbirdChainID) == 0:
+		return blockTime.Cmp(songbirdStateConnectorActivationTime) >= 0
+	default:
+		return false
+	}
+}
+
+func isTestingChain(chainID *big.Int) bool {
+	return chainID.Cmp(flareChainID) != 0 && chainID.Cmp(songbirdChainID) != 0
+}
+
+func submitAttestationSelector(chainID *big.Int, blockTime *big.Int) []byte {
+	switch {
+	default:
+		return []byte{0xcf, 0xd1, 0xfd, 0xad}
+	}
```

Comment:
No point in having a switch here. It can be added later if needed but here it just makes the code less clear for no good  reason.